module_test.go

```diff
@@ -241,6 +244,85 @@ func TestModuleSuccess(t *testing.T) {
 		)
 		require.NoError(t, app.Err())
 	})
+
+	t.Run("custom logger for module", func(t *testing.T) {
+		t.Parallel()
+
+		var spy fxlog.Spy
+
+		redis := fx.Module("redis",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "redis"}
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+			fx.NopLogger,
+		)
+
+		app := fxtest.New(t,
+			redis,
+			fx.Supply(&spy),
+			fx.WithLogger(func(spy *fxlog.Spy) fxevent.Logger {
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		// events from module with a custom logger not logged in app logger
+		assert.Equal(t, []string{
+			"Supplied", "Provided", "Provided", "Provided",
+			"LoggerInitialized", "Invoking", "Invoked",
+		}, spy.EventTypes())
+
+		spy.Reset()
+		app.RequireStart().RequireStop()
+
+		require.NoError(t, app.Err())
+
+		assert.Equal(t, []string{"Started", "Stopped"}, spy.EventTypes())
+	})
+
+	t.Run("Not using a custom logger for module defaults to app logger", func(t *testing.T) {
+		t.Parallel()
+
+		var spy fxlog.Spy
+
+		redis := fx.Module("redis",
+			fx.Provide(func() *Logger {
+				return &Logger{Name: "redis"}
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		app := fxtest.New(t,
+			redis,
+			fx.Supply(&spy),
+			fx.WithLogger(func(spy *fxlog.Spy) fxevent.Logger {
+				return spy
+			}),
+			fx.Invoke(func(r *Logger) {
+				assert.Equal(t, "redis", r.Name)
+			}),
+		)
+
+		// events from module also logged in app logger
+		assert.Equal(t, []string{
+			"Supplied", "Provided", "Provided", "Provided", "Provided",
+			"LoggerInitialized", "Invoking", "Invoked", "Invoking", "Invoked",
+		}, spy.EventTypes())
+
+		spy.Reset()
+		app.RequireStart().RequireStop()
+
+		require.NoError(t, app.Err())
+
+		assert.Equal(t, []string{"Started", "Stopped"}, spy.EventTypes())
```

Comment:
These tests may actually be better off written as a table test. 
See: https://github.com/uber-go/guide/blob/master/style.md#test-tables